docs/javascript-unit-tests.md

```diff
@@ -0,0 +1,712 @@
+# Writing JavaScript Unit Tests
+
+This page provides instructions for writing
+a new JavaScript unit test in cfgov-refresh.
+
+[Jest](https://jestjs.io/docs/en/getting-started) is the framework we use
+for writing and running JavaScript unit tests.
+If you’re not familiar with it,
+it would be a good idea to peruse their docs before diving in here.
+
+For documentation on _running_ tests,
+[see the main front-end testing page](../testing-fe/#unit-testing).
```

Comment:
Two things: first, it kinda bothers me that this document distinguishes between itself and that one as "the main front-end testing page". Why not rename that page so it's explicitly about running tests? But the second thing is, at the bottom of this document, it has how to run the tests too. Let's consolidate that information to... somewhere.

Reply:
Yeah, this is something I intended to resolve by the end of this PR and sorta forgot about. I'll look at it again.

Reply:
I actually wonder — the other document seems almost entirely about browser testing. Maybe we rename it to "Browser Testing"?

Reply:
Yeah, that's what I was leaning toward – a Browser Testing page and a JavaScript Unit Testing page, each of which explains how to run those kinds of tests.

Reply:
Maybe add `aka acceptance tests` or similar to the browser testing page, since as to my understanding that's what they are; they are the full system running in a browser and being tested. Also the command is `gulp test:acceptance`. Browser tests is like calling the unit tests "command-line tests." It's helpful to have "browser tests" there to convey what's happening, but having the technical name would be good too so there's a shared term across the tooling and docs.